crates/primitives/src/sorted_vec.rs

```diff
@@ -41,12 +41,26 @@ impl<T: Ord + Clone> SortedVec<T> {
 
     /// Checks if the [`SortedVec`] contains the given value.
     pub fn contains(&self, value: &T) -> bool {
-        self.binary_search(value).is_ok()
+        self.binary_search(value).is_some()
     }
 
     /// Perform binary search on the vector
-    pub fn binary_search(&self, value: &T) -> Result<usize, usize> {
-        self.inner.binary_search(value)
+    pub fn binary_search(&self, value: &T) -> Option<&T> {
+        self.inner
+            .binary_search(value)
+            .ok()
+            .map(|idx| &self.inner[idx])
+    }
```

Comment:
As a sanity check, this fn doesn't *really* make sense.  It's just returning a ref to a different instance of the same type.  We'd *expect* this to be literally equivalent based on the relationships implied by the `Eq` and `Ord` impls, but we're not.  Instead (for this to be useful) we'd construct a different dummy instance of `T`.

Using `_by_key` is the interpretation here that really makes sense since it doesn't rely on this weird implication.

Reply:
Fixed.